Brokerages/Oanda/OandaBrokerage.cs

```diff
@@ -0,0 +1,1052 @@
+                        requestParams.Add("lowerBound", ((StopMarketOrder)order).StopPrice.ToString(CultureInfo.InvariantCulture));
+                        break;
+                }
+
+                //3 months is the max expiry for Oanda, and OrderDuration.GTC is only currently available
+                requestParams.Add("expiry", XmlConvert.ToString(DateTime.Now.AddMonths(3), XmlDateTimeSerializationMode.Utc));
+            }
+        }
+
+
+        /// <summary>
+        /// Checks for fill events by registering to the event session to receive events.
+        /// </summary>
+        private void CheckForFills()
+        {
+            // reentrance gaurd
+            if (!Monitor.TryEnter(_fillLock))
+            {
+                return;
+            }
+            try
+            {
+                var session = new EventsSession(AccountId);
+                session.DataReceived += OnEventReceived;
+                session.StartSession();
+            }
+            finally
+            {
+                Monitor.Exit(_fillLock);
+            }
+        }
+
+        private void OnEventReceived(Event data)
+        {
+            #if DEBUG
+            Console.Out.Write("---- On Event Received ----");
+            Console.Out.Write(data.transaction);
+            #endif
+            
+            if (data.transaction != null)
+            {
+                if (data.transaction.type == "ORDER_FILLED")
+                {
+                    var qcOrder = _orderProvider.GetOrderByBrokerageId(data.transaction.orderId);
+                    var fill = new OrderEvent(qcOrder, "Oanda Fill Event")
+                    {
+                        Status = OrderStatus.Filled,
+                        //    // this is guaranteed to be wrong in the event we have multiple fills within our polling interval,
+                        //    // we're able to partially cope with the fill quantity by diffing the previous info vs current info
+                        //    // but the fill price will always be the most recent fill, so if we have two fills with 1/10 of a second
+                        //    // we'll get the latter fill price, so for large orders this can lead to inconsistent state
+                        FillPrice = (decimal) data.transaction.price,
+                        FillQuantity = data.transaction.units
+                    };
+
+                    // flip the quantity on sell actions
+                    if (qcOrder.Direction == OrderDirection.Sell)
+                    {
+                        fill.FillQuantity *= -1;
+                    }
+                    OnOrderEvent(fill);
+                }
+            }
+        }
+
+        /// <summary>
+        /// Obtain the active open Trade List from Oanda.
+        /// </summary>
+        /// <param name="requestParams"></param>
+        /// <returns></returns>
+        public TradesResponse GetTradeList(Dictionary<string, string> requestParams = null)
```

Comment:
various API wrapping methods accepts a dictionary of parameters, it would be nice to define them all as formal parameters and make the dictionary internally in the method